programs/uxd/src/instructions/credix_lp/exchange_liquidity_with_credix_lp_depository.rs

```diff
@@ -0,0 +1,319 @@
+    )?;
+    let credix_lp_depository_shares_amount_decrease = compute_decrease(
+        credix_lp_depository_shares_amount_before,
+        credix_lp_depository_shares_amount_after,
+    )?;
+    let user_collateral_amount_decrease =
+        compute_decrease(user_collateral_amount_before, user_collateral_amount_after)?;
+    let receiver_credix_shares_amount_increase = compute_increase(
+        receiver_credix_shares_amount_before,
+        receiver_credix_shares_amount_after,
+    )?;
+
+    // Log deltas for debriefing the changes
+    msg!(
+        "[identity_depository_collateral_amount_increase:{}]",
+        identity_depository_collateral_amount_increase
+    );
+    msg!(
+        "[credix_lp_depository_shares_amount_decrease:{}]",
+        credix_lp_depository_shares_amount_decrease
+    );
+    msg!(
+        "[user_collateral_amount_decrease:{}]",
+        user_collateral_amount_decrease
+    );
+    msg!(
+        "[receiver_credix_shares_amount_increase:{}]",
+        receiver_credix_shares_amount_increase
+    );
+
+    // Fact check
+    require!(
+        identity_depository_collateral_amount_increase == collateral_amount,
+        UxdError::CollateralDepositAmountsDoesntMatch
+    );
+    require!(
+        credix_lp_depository_shares_amount_decrease == exchanged_shares_amount,
+        UxdError::CollateralDepositDoesntMatchTokenValue
+    );
+    require!(
+        user_collateral_amount_decrease == collateral_amount,
+        UxdError::CollateralDepositAmountsDoesntMatch
+    );
+    require!(
+        receiver_credix_shares_amount_increase == exchanged_shares_amount,
+        UxdError::CollateralDepositDoesntMatchTokenValue
+    );
+
+    // ---------------------------------------------------------------------
+    // -- Phase 4
+    // -- Emit resulting event, and update onchain accounting
+    // ---------------------------------------------------------------------
+
+    // Emit event
+    emit!(ExchangeLiquidityWithCredixLpDepositoryEvent {
+        controller_version: ctx.accounts.controller.load()?.version,
+        depository_version: ctx.accounts.credix_lp_depository.load()?.version,
+        controller: ctx.accounts.controller.key(),
+        depository: ctx.accounts.credix_lp_depository.key(),
+        collateral_amount,
+        shares_amount: exchanged_shares_amount,
+    });
+
+    // Accounting for identity_depository
+    let mut identity_depository = ctx.accounts.identity_depository.load_mut()?;
+    identity_depository.collateral_amount_deposited = checked_add(
+        identity_depository.collateral_amount_deposited,
+        collateral_amount.into(),
+    )?;
+    identity_depository.redeemable_amount_under_management = checked_add(
+        identity_depository.redeemable_amount_under_management,
+        collateral_amount.into(),
+    )?;
+
+    // Accounting for credix_lp_depository
+    let mut credix_lp_depository = ctx.accounts.credix_lp_depository.load_mut()?;
```

Comment:
Note: the precision loss occuring on top, increase the value per share

Reply:
Yep, by a very small amount. That amount will show up in the "Profits to be collected" automatically, I don't think we need to do anything about it.

Reply:
Resolving this for now, I don't have any action item here.